File: internal/utils/strings.go
package utils

import "strings"

// ParseCSV splits a comma-separated string and returns trimmed non-empty values.
// Returns nil for empty/whitespace-only input.
// This function is used throughout the codebase to parse comma-separated
// industry and geography values stored in the database.
func ParseCSV(s string) []string {
	if s == "" {
		return nil
	}

	var result []string
	for _, v := range strings.Split(s, ",") {
		trimmed := strings.TrimSpace(v)
		if trimmed != "" {
			result = append(result, trimmed)
		}
	}

	if len(result) == 0 {
		return nil
	}

	return result
}

File: sentinel/utils/strings.py
"""String utilities."""


def parse_csv_field(value: str | None) -> list[str]:
    """Parse a comma-separated string into a list of stripped, non-empty values.

    Args:
        value: Comma-separated string, or None/empty

    Returns:
        List of stripped non-empty strings
    """
    if not value:
        return []
    return [item.strip() for item in value.split(",") if item.strip()]
